### DISCUSSION TITLE
Seeking Guidance on Integrating Next.js Default i18n Setup with Fumadocs i18n Documentation

### DISCUSSION THREAD

**Discussion by smilewilson1999** [fuma-nama/fumadocs#1146] [Q&A]
I am using Next.js with its recommended default i18n configuration (https://nextjs.org/docs/14/app/building-your-application/routing/internationalization#resources), where internationalized routes are dynamically handled through [lang] and middleware redirects. Despite thorough research and reviewing all available online resources and documentation, I am still unsure how to effectively integrate it with Fumadocs's i18n documentation.

This is my middleware.ts:
<img width="376" alt="image" src="https://github.com/user-attachments/assets/f24292b8-92e8-44bf-a30c-600a85f67d2c">

I would greatly appreciate any guidance, examples, or insights on how to properly configure both systems to work together.

**Comment by fuma-nama** [comment]
https://fumadocs.vercel.app/docs/ui/internationalization

**Reply by smilewilson1999** [reply]
I just got stuck!! After a good sleep, everything goes smoothly. Thank you.
